### ISSUE TITLE
Changing project directory to (macOS) alias of same name breaks app

### ISSUE THREAD

**Issue by paced** [desktop/desktop#2713]
### Description

I had a directory, `Projects`, that contained all of my GitHub project directories. I recently needed to move that directory to another drive, but wanted to alias it so I wouldn't get confused. I kept the name of the alias the same as the directory. Upon opening GitHub Desktop, I would get a non-graceful `Error: spawn ENOTDIR`. It was not possible to remove the repository.

This was fixed by changing the name of the alias, but I figured maintainers might want to, at the very least, change the error message. If Desktop can follow aliases, it can't when a folder once was directly in a directory.

### Version

**GitHub Desktop version:** 0.8.2-77b74af1

**OS version:** macOS Sierra 10.12.6 (16G29)

### Steps to Reproduce

1. Clone a repository into `/some/random/path/Repository` from GitHub Desktop.
2. Close GitHub Desktop and move the repository, adding an alias through the context menu: `/some/random/path/Repository`.
3. Open GitHub Desktop.

**Expected behavior:** Either stating that the repository was not found, or following the alias/symlink to the new repository.

**Actual behavior:** `Error: spawn ENOTDIR`.

**Reproduces how often:** 100%.

### Logs

```
2017-09-13T07:41:45.145Z - error: [ui] Error performing periodic fetch
Error: spawn ENOTDIR
    at exports._errnoException (util.js:1050:11)
    at ChildProcess.spawn (internal/child_process.js:319:11)
    at exports.spawn (child_process.js:390:9)
    at Object.exports.execFile (child_process.js:155:15)
    at Object.module.(anonymous function) [as execFile] (ELECTRON_ASAR.js:200:20)
    at /webpack:/app/node_modules/dugite/build/lib/git-process.js:65:1
    at Promise (<anonymous>)
    at Function.exec (/webpack:/app/node_modules/dugite/build/lib/git-process.js:48:1)
    at a.measure (/webpack:/app/src/lib/git/core.ts:113:13)
    at Object.t.measure (/webpack:/app/src/ui/lib/git-perf.ts:22:10)
2017-09-13T07:42:12.530Z - info: [ui] Executing installGlobalLFSFilter: git lfs install --skip-repo (took 1.358s)
```

#### Additional Information

None that I can think of.

**Comment by shiftkey** [comment]
> Close GitHub Desktop and move the repository, adding an alias: `/some/random/path/Repository`.

Can you clarify the symlinking you're doing here, preferable with example commands, to ensure we're able to recreate the issue in the right way?

**Comment by paced** [comment]
Sure, but it looks like my understanding of "shortcuts" in macOS differs to that of traditional symbolic links and hard links. What I meant to say is "alias", which can be recreated by right clicking on a file in Finder and selecting "make alias."

**Comment by joshaber** [comment]
Huh, yeah, it's not terribly surprising that we're not resolving aliases, but we definitely should be.

**Comment by paced** [comment]
No worries. I should note that aliases normally work (though I'm not too familiar with the macOS filesystem). Only when the directory is moved and an alias is placed where the directory used to be does this error occur.

Thanks for working on some pretty useful software 😄 !

**Comment by shiftkey** [comment]
This is a limitation of Git, because Git isn't able to resolve an alias in the same way it can a symlink. From [Wikipedia](https://en.wikipedia.org/wiki/Alias_(Mac_OS)#Relation_to_BSD_symbolic_and_hard_links):

> However, when using the shell command line, macOS aliases are not recognized: for example, you cannot use the cd command with the name of an alias file. **This is because an alias is implemented as a file on the disk that must be interpreted by an API while links are implemented within the filesystem and are thus functional at any level of the OS.**

I didn't find much in the way of other users complaining about this issue when googling, but that might be because `alias` is already a Git concept. This is also a discussion to have with the larger  [Git community](https://git.wiki.kernel.org/index.php/GitCommunity), but I suspect they'd be hesitant to add support for this without significant interest and willing contributors to get involved.

**Comment by a-dpq** [comment]
I just ran into this very same issue, and it completely blocks the user from using GitHub Desktop by presenting repeated error messages without any helpful information. The only way I was able to resolve it was by finding this writeup. @shiftkey, are you sure this is a Git issue? It seems like it might be a problem with GitHub Desktop.

**Comment by shiftkey** [comment]
@a-dpq have you been able to use this repository using the Git command line? I'm not aware of any support in Git itself for aliases on macOS, but I haven't had the bandwidth to see this with my own eyes...

**Comment by a-dpq** [comment]
Yes, I was able to use Git on the repository via the command line – though I didn't try to access it using the (macOS) alias. The issue here, as it seems to me, is that GitHub Desktop is getting confused by the situation that @paced described above, where a repository used to be located in a folder, but now there is an alias with the same name there. In this situation, GitHub Desktop produces a repeated error message that blocks the user from using the application. I was only able to resolve it by renaming the alias in the folder and then selecting the correct location of the repository in GitHub Desktop.

**Comment by a-dpq** [comment]
I should add that perhaps the repeated error messages were due to having multiple aliases of repositories in the folder. In any case, the error message is cryptic.

**Comment by shiftkey** [comment]
> In any case, the error message is cryptic.

@a-dpq the error message is deep in the bowels of NodeJS and it's handling of the aborted `spawn` command. I can't change that, but maybe I can improve our error handling.

To investigate how our error handling is behaving, we need a complete set of steps to reproduce this behaviour, ideally with command line instructions that can be run by someone else. Without those details, we could sink a lot of time and not end up looking at the same behaviour.

If you can help out with that, making the error message more friendly becomes achievable.

**Comment by a-dpq** [comment]
I just reproduced the error with these steps. This assumes that you have at least one local repository tracked by GitHub Desktop:

1. Move the repository to a different directory.
2. Make an alias in the Finder (command L) of the repository folder, and move it to the repository's previous location.
3. Remove " alias" from the file name of the alias.
4. Open GitHub Desktop.

Expected behavior: Repeated error messages "spawn ENOTDIR" that completely block the user from using the application.

I just verified that the error messages are repeated and block any use of the application until the problem is resolved as described in the posts above.
